packages/react/src/modules/feed/components/NotificationCell/NotificationCell.tsx

```diff
@@ -23,91 +35,129 @@ type BlockByName = {
   [name: string]: ContentBlock;
 };
 
+function maybeNavigateToUrlWithDelay(url?: string) {
+  if (url && url !== "") {
+    setTimeout(() => window.location.assign(url), 200);
+  }
+}
+
 export const NotificationCell = React.forwardRef<
   HTMLDivElement,
   NotificationCellProps
->(({ item, onItemClick, avatar, children, archiveButton }, ref) => {
-  const { feedClient, colorMode } = useKnockFeed();
-  const { locale } = useTranslations();
-
-  const blocksByName: BlockByName = useMemo(() => {
-    return item.blocks.reduce((acc, block) => {
-      return { ...acc, [block.name]: block };
-    }, {});
-  }, [item]);
-
-  const actionUrl = blocksByName.action_url && blocksByName.action_url.rendered;
-
-  const onClick = React.useCallback(() => {
-    // Mark as interacted + read once we click the item
-    feedClient.markAsInteracted(item);
-
-    if (onItemClick) return onItemClick(item);
-
-    // Delay when we navigate, until we've actually issued our API call.
-    setTimeout(() => {
-      if (actionUrl && actionUrl !== "") {
-        window.location.assign(actionUrl);
-      }
-    }, 200);
-  }, [item]);
-
-  const onKeyDown = React.useCallback(
-    (ev: React.KeyboardEvent<HTMLDivElement>) => {
-      switch (ev.key) {
-        case "Enter": {
-          ev.stopPropagation();
-          onClick();
-          break;
+>(
+  (
+    { item, onItemClick, onButtonClick, avatar, children, archiveButton },
+    ref,
+  ) => {
+    const { feedClient, colorMode } = useKnockFeed();
+    const { locale } = useTranslations();
+
+    const blocksByName: BlockByName = useMemo(() => {
+      return item.blocks.reduce((acc, block) => {
+        return { ...acc, [block.name]: block };
+      }, {});
+    }, [item]);
+
+    const actionUrl = (blocksByName.action_url as TextContentBlock)?.rendered;
+    const buttonSet = blocksByName.actions as ButtonSetContentBlock;
+
+    const onContainerClickHandler = React.useCallback(() => {
+      // Mark as interacted + read once we click the item
+      feedClient.markAsInteracted(item);
+
+      if (onItemClick) return onItemClick(item);
+
+      return maybeNavigateToUrlWithDelay(actionUrl);
+    }, [item, actionUrl, onItemClick, feedClient]);
+
+    const onButtonClickHandler = React.useCallback(
+      (e: React.MouseEvent, button: ButtonBlock) => {
+        feedClient.markAsInteracted(item);
+
+        if (onButtonClick) return onButtonClick(item, button);
+
+        return maybeNavigateToUrlWithDelay(button.action);
```

Comment:
Just double checking - is button's `action` expected to be a fully rendered url here? Asking because `ButtonBlock` does not have a `rendered` field, which gave me the impression they aren't liquid rendered.

Reply:
Yes, it's absolutely the rendered value here. You can see in the feed template renderer for the NotificationEngine where it's doing this. We don't send down `rendered` as it doesn't make sense in the context of these buttons to pass the original liquid template strings and the rendered variables, so we just send down the result.

Reply:
Got it, thanks for clarifying!